### PR TITLE
(chapter 7): Add derivation of loss function based on Bradley Terry model of preferences

### PR DESCRIPTION
**Pull request by zafstojano**
Another nitpick regarding the explanation of the loss function derived from Bradley-Terry's model of preferences. 

The current version says:
old

I don't believe it is the case that "by taking the gradient (of the preference function) with respect to the model parameters, we can arrive at the loss function to train a reward model"

In contrast, I think that we arrive at the loss function to train a reward model by maximizing (log) likelihood of the preferences being ordered correctly (`P(y_w > y_l)`). Or analogously, by minimizing the negative log-likelihood, which turns out to be the loss!

new

Personally, I found I understood the loss much better with the derivation. If in turn you feel like the math breaks the reading flow or clutters the page, I can move it to an appendix at the end.

### CODE REVIEW

**Comment by natolambert** [comment]
Nice! I haven't looked at this derivation in a wild, is good to add. May reformat it at some point, but definitely a useful addition.
